src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const splittedDate = date.split(fromFormat[3]);
+  const oldDateObj = {};
+  const result = [];
+
+  for (let i = 0; i < splittedDate.length; i++) {
+    oldDateObj[fromFormat[i]] = splittedDate[i];
+    // make new object via input date;
+  }
+  changeYearsFormat(oldDateObj, toFormat);
```

Comment:
The function `changeYearsFormat` mutates its first argument. It's a good practice to avoid mutating input parameters as it can lead to unexpected bugs. Consider returning a new object with the updated values instead.